### PR TITLE
chore: restore pty tests and make them run on the Linux CI

### LINKED ISSUE #18129: `integration::repl` tests hang on local machine
Source: https://discord.com/channels/684898665143206084/684911491035430919/1083922631268503622

Repro (main branch):

```bash
cargo test integration::repl
...
test integration::repl::variable ... ok
test integration::repl::pty_assign_global_this has been running for over 60 seconds
test integration::repl::pty_bad_input has been running for over 60 seconds
test integration::repl::pty_clear_function has been running for over 60 seconds
test integration::repl::pty_complete_declarations has been running for over 60 seconds
test integration::repl::pty_complete_expression has been running for over 60 seconds
test integration::repl::pty_complete_imports has been running for over 60 seconds
test integration::repl::pty_complete_imports_no_panic_empty_specifier has been running for over 60 seconds
test integration::repl::pty_complete_primitives has been running for over 60 seconds
test integration::repl::pty_complete_symbol has been running for over 60 seconds
test integration::repl::pty_emoji has been running for over 60 seconds
test integration::repl::pty_ignore_symbols has been running for over 60 seconds
test integration::repl::pty_null has been running for over 60 seconds
test integration::repl::pty_syntax_error_input has been running for over 60 seconds
test integration::repl::pty_tab_handler has been running for over 60 seconds
test integration::repl::pty_unpaired_braces has been running for over 60 seconds
```

### PR DESCRIPTION
**Pull request by dsherret**
~~The repl tests now run on my system again. I will test on linux in a second.~~ The tests didn't run on linux, so I'm rewriting them.

This does a few things:

1. Rewrites the tests to be more back and forth rather than getting the output all at once (which I believe was causing the hangs on linux and maybe mac)
2. Runs the pty tests on the linux ci.
3. Fixes a bunch of tests that were just wrong.
4. Adds timeouts on the pty tests.

~~Closes #18129~~ - Hangs on mac

### CODE REVIEW

File: cli/tools/repl/editor.rs [hunk 1]
```diff
@@ -249,69 +249,79 @@ fn validate(input: &str) -> ValidationResult {
   let mut in_template = false;
   let mut div_token_count_on_current_line = 0;
   let mut last_line_index = 0;
-
-  for item in deno_ast::lex(input, deno_ast::MediaType::TypeScript) {
-    let current_line_index = line_info.line_index(item.range.start);
+  let mut queued_validation_error = None;
```
Comment (dsherret) [inline cli/tools/repl/editor.rs:252]: The change here fixes another regression due to swc not having regex literals as tokens anymore. A pty test pointed this out.

File: test_util/src/builders.rs [hunk 2]
```diff
@@ -511,33 +539,30 @@ impl TestCommandOutput {
       .expect("call .split_output() on the builder")
   }
 
+  #[track_caller]
```
Comment (dsherret) [inline test_util/src/builders.rs:542]: See https://blog.rust-lang.org/2020/08/27/Rust-1.46.0.html#track_caller

I just learned about this and it works way better than the `failed_position()` thing.

Comment (bartlomieju) [reply test_util/src/builders.rs:542]: Nice find

**Review by bartlomieju — APPROVED** [review]
LGTM, sadly the tests don't yet work properly on macOS. We tried several things but none of them worked so far. I'll try to revisit this topic later this week.

**Pull request merged**